api/src/main/java/com/github/retrooper/packetevents/util/PacketEventsImplHelper.java

```diff
@@ -103,7 +107,7 @@ public static PacketReceiveEvent handleServerBoundPacket(Object channel, User us
                 task.run();
             }
         }
-        return packetReceiveEvent;
+        return buffer;
```

Comment:
Why is the buffer returned, and the method return changed?

Reply:
Because we have to retain the correct buffer, previously the buffer needed to be retained was always the `input` one, now if the packet is marked for `re-encode`, we clear the `input `buffer and allocate a new one where to write data.